group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorShard.java

```diff
@@ -400,6 +408,14 @@ public CoordinatorResult<OffsetDeleteResponseData, Record> deleteOffsets(
         return offsetMetadataManager.deleteOffsets(request);
     }
 
+    public CoordinatorResult<Void, Record> cleanupGroupMetadata() {
+        List<Record> records = new ArrayList<>();
+        offsetMetadataManager.cleanupOffsetMetadata(records, config.offsetsRetentionMs);
+        groupMetadataManager.cleanupGroupMetadata(records);
+
+        return new CoordinatorResult<>(records, null);
```

Comment:
The number of records here is potentially unlimited. Therefore, we could hit the max message size. If we do, the expiration process would basically fail and never recover. Have you thought about how we could handle this?